app/services/generate_bill_run.service.js

```diff
@@ -37,18 +48,62 @@ class GenerateBillRunService {
     if (!billRun.$editable()) {
       throw Boom.badData(`Bill run ${billRun.id} cannot be edited because its status is ${billRun.status}.`)
     }
+
+    if (billRun.$empty()) {
+      throw Boom.badData(`Summary for bill run ${billRun.id} cannot be generated because it has no transactions.`)
+    }
   }
 
   static async _setGeneratingStatus (billRun) {
     await billRun.$query()
       .patch({ status: 'generating' })
   }
 
-  static async _summariseBillRun (billRun) {
-    return BillRunModel.transaction(async trx => {
-      await this._summariseZeroValueInvoices(billRun, trx)
-      await this._summariseDeminimisInvoices(billRun, trx)
-    })
+  static async _saveTransactions (transactions, trx) {
+    const allSavedTransactions = []
```

Comment:
I might be wrong, but it does not look like we are actually using the returned array.

So, in our efforts to shave every millisecond we can I'd suggest

- drop collecting the created transactions in an array
- drop `.returning(*)` from the `query()`

When we create a single record this is an efficient way to get the created data back for use in the response. But in this case, where we don't need the data, we might get a slight performance bump from not asking PG to generate a response, and less traffic going across the network.